script.js

```diff
@@ -0,0 +1,90 @@
+const myLibrary = [];
+
+function addRemoveButton() {
+  const remove = document.querySelector('.remove');
+  const trd = document.createElement('tr');
+  const tdd = document.createElement('td');
+  const button = document.createElement('button');
+  button.textContent = 'remove';
+  button.className = 'remove_button';
+  button.setAttribute('id', myLibrary[myLibrary.length - 1].title);
+  button.setAttribute('onclick', `removeFunction(${button.id})`);
+  tdd.appendChild(button);
+  trd.appendChild(tdd);
+  remove.appendChild(trd);
+}
+
+function displayBook() {
+  const table = document.querySelector('table');
+  const tr = document.createElement('tr');
+  const title = document.createElement('td');
+  const author = document.createElement('td');
+  const pages = document.createElement('td');
+  const read = document.createElement('td');
+  title.textContent = myLibrary[myLibrary.length - 1].title;
+  author.textContent = myLibrary[myLibrary.length - 1].author;
+  pages.textContent = myLibrary[myLibrary.length - 1].pages;
+  read.textContent = myLibrary[myLibrary.length - 1].read;
+  const updateButton = document.createElement('button');
+  const readId = `${myLibrary[myLibrary.length - 1].title}read`;
+  read.setAttribute('id', readId);
+  updateButton.setAttribute('onclick', `updateStatus('${myLibrary[myLibrary.length - 1].title}')`);
+  updateButton.textContent = 'update';
+  read.appendChild(updateButton);
+  tr.className = (title.textContent);
+  tr.appendChild(title);
+  tr.appendChild(author);
+  tr.appendChild(pages);
+  tr.appendChild(read);
+  table.appendChild(tr);
+  addRemoveButton();
+}
+
+const addBookButton = document.querySelector('button');
+addBookButton.onclick = function () {
+  const book = {
+    title: prompt('Enter title'),
+    author: prompt('Enter Author'),
+    pages: prompt('Number of pages'),
```

Comment:
- [ ] As discussed above, avoid using` prompt`. Use modals or forms instead.

Reply:
Hello I would like to bring some things to the attention 
I believe that my app is more efficient than what is suggested as a way to solve it in the instructions and it does the same thing in a better way.

In my app every time a user adds a book it adds it to the array and then it adds a line to the table.

if I were to create a function that loops over the array every time someone adds a book this would be much less efficient.

the loop will need to run from the beginning till the end and the table will be reconstructed as well just for adding one book.


for that reason instead of having two functions which one is called 'addBookToLibrary 'and another is called 'displayBooks'

  
I only have one function which is called 'displayBook' but actually adds the books but also displays it.